src/extensions/averagewaveforms/AverageWaveformsView/setupElectrodes.ts

```diff
@@ -102,7 +102,22 @@ const setupVerticalElectrodes = ({width, height, electrodeIds}: {width: number,
     }
 }
 
-const setupElectrodes = (args: {width: number, height: number, electrodeLocations: Vec2[], electrodeIds: number[], layoutMode: 'geom' | 'vertical'}): {
+const fixDegenerateCase = (electrodeLocations: Vec2[]) => {
+    const box = {
+        xmin: getArrayMin(electrodeLocations.map(e => (e[0]))),
+        xmax: getArrayMax(electrodeLocations.map(e => (e[0]))),
+        ymin: getArrayMin(electrodeLocations.map(e => (e[1]))),
+        ymax: getArrayMax(electrodeLocations.map(e => (e[1])))
+    }
+    if ((box.xmin === box.xmax) && (box.ymin === box.ymax)) {
```

Comment:
So this still works okay if one of the dimensions is zero? (I see it's mapped to a linear geometry, but I was also wondering what would happen if the electrodes were actually laid out in a line anyway)

Reply:
I don't think it's a problem if they are laid out linearly. The estimated spacing is then non-zero.